frontends/web/src/routes/buy/pocket.tsx

```diff
@@ -0,0 +1,144 @@
+import { signAddr, getPocketURL } from '../../api/backend';
+import { Header } from '../../components/layout';
+import { Spinner } from '../../components/spinner/Spinner';
+import { useLoad } from '../../hooks/api';
+import Guide from './guide';
+import style from './iframe.module.css';
+
+interface PocketProps {
+    code: string;
+}
+
+type Props = PocketProps & TranslateProps;
+
+// TODO:
+// - add T&C
+const PocketComponent: FunctionComponent<Props> = ({ code, t }) => {
+  const [height, setHeight] = useState(0);
+  const iframeURL = useLoad(getPocketURL(code));
+
+  const ref = createRef<HTMLDivElement>();
+  const iframeRef = createRef<HTMLIFrameElement>();
+  var resizeTimerID: any = undefined;
+
+  const name = 'Bitcoin';
+
+  useEffect(() => {
+    onResize();
+    window.addEventListener('resize', onResize);
+    window.addEventListener('message', onMessage);
+
+    return () => {
+      window.removeEventListener('resize', onResize);
+      window.removeEventListener('message', onMessage);
+    };
+  });
+
+  const onResize = () => {
+    if (resizeTimerID) {
+      clearTimeout(resizeTimerID);
+    }
+    resizeTimerID = setTimeout(() => {
+      if (!ref.current) {
+        return;
+      }
+      setHeight(ref.current.offsetHeight);
+    }, 200);
+  };
+
+  const onMessage = (e: MessageEvent) => {
+    try {
+      const message = parseMessage(e.data);
+      console.log(JSON.stringify(message, undefined, 2));
+
+      if (!code) {
+        return;
+      }
+
+      // handle address request from moonpay
+      if (message.type === V0MessageType.RequestAddress && message.withMessageSignature) {
+        signAddr(
+          message.withScriptType ? message.withScriptType : 'p2wpkh',
+          String(message.withMessageSignature),
+          code)
+          .then( resp => {
+            console.log(JSON.stringify(resp));
+
+            if (!resp.addr || !resp.sig ) {
+              console.log('Something went wrong!');
+            } else {
+              sendAddress(resp.addr, resp.sig);
+            }
+          });
+      }
+    } catch (e) {
+      // ignore messages that could not be parsed
+      // probably not intended for us, anyway
+    }
+  };
+
+  const sendAddress = (address: string, sig: string) => {
+    const { current } = iframeRef;
+
+    if (!current) {
+      return;
+    }
+
+    const message = serializeMessage({
+      version: MessageVersion.V0,
+      type: V0MessageType.Address,
+      bitcoinAddress: address,
+      signature: sig,
+    });
+
+    console.log(message);
+    current.contentWindow?.postMessage(message, '*');
+  };
+
+  return (
+    <div className="contentWithGuide">
+      <div className="container">
+        <div className={style.header}>
+          <Header title={<h2>{t('buy.info.title', { name })}</h2>} />
+        </div>
+        <div ref={ref} className="innerContainer">
+          <div className="noSpace" style={{ height }}>
```

Comment:
slightly confused, what is double braces? and isn't `height` an int not a style?

Reply:
TIL!